### PR TITLE
UCP/TEST: Avoid calling progress from probe, recv, and msg_recv.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1435/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2082/ for details.

**Comment by yosefe** [comment]
@rosenbaumalex @rafiw
@shamisp pls review
I've checked and it doesn't impact UCP tag message rate

**Comment by rafiw** [comment]
this brakes accelio data flow, not sure why but it looks like accelio is stuck and the epoll is not getting invoked

File: src/ucp/api/ucp.h [hunk 1]
```diff
@@ -636,6 +636,10 @@ void ucp_worker_release_address(ucp_worker_h worker, ucp_address_t *address);
  * @li Transport layers, implementing asynchronous progress using threads,
  * require callbacks and other user code to be thread safe.
  *
+ * @note
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: s/@note/@li

and remove the space

File: src/ucp/api/ucp.h [hunk 2]
```diff
@@ -636,6 +636,10 @@ void ucp_worker_release_address(ucp_worker_h worker, ucp_address_t *address);
...
+ * progress can be done implicitly by blocking functions, and will not be done
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: The state of communication can be advanced (progressed) by blocking routines. Nevertheless, the non-blocking routines can not be used for communication progress

File: src/ucp/api/ucp.h [hunk 3]
```diff
@@ -1151,14 +1155,20 @@ ucs_status_ptr_t ucp_tag_recv_nb(ucp_worker_h worker, void *buffer, size_t count
...
  * @param [out] info        If the matching message is found the descriptor is
  *                          filled with the details about the message.
  *
  * @return NULL                      - No match found.
  * @return Message handle (not NULL) - If message is matched the message handle
- *                                   is returned.
+ *                                     is returned.
  *
- * @todo Clarify release logic for remote == 0.
+ * @note This function does not make progress on the network. If this function
```
Comment (shamisp) [inline src/ucp/api/ucp.h]: s/function/routine

Comment (shamisp) [inline src/ucp/api/ucp.h]: I would proposed to use similar language. Progress = advance communication state.

**Comment by shamisp** [comment]
Technical side look good. Some comments for the documentation

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1467/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2115/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1469/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2117/ for details.

**Comment by yosefe** [comment]
@shamisp pls take a look

**Comment by shamisp** [comment]
👍

**Pull request merged**